### PR TITLE
Asians

### PR DESCRIPTION
**Pull request by vargala4**

### CODE REVIEW

**Review by erno-solymosi — APPROVED** [review]

File: src/contract.py [hunk 1]
```diff
@@ -184,8 +186,9 @@ def payoff(self, spot: dict[float, float]) -> float:
         if not set(timeline).issubset(set(spot.keys())):
             self.raise_missing_spot_error(list(spot.keys()))
         obs = [spot[t] for t in timeline]
-        in_out = self.barrier.in_out
-        is_breached = int(self.barrier.is_breached(obs))
+        in_out = self._barrier.get_in_out()
```
Comment (friedgab) [inline src/contract.py:189]: Why access modifiers have been introduced again?

File: src/pricer.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from src.model import *
 from src.numerical_method import *
 
+maci = 12
```
Comment (friedgab) [inline src/pricer.py:12]: What is this global variable for?

File: src/pricer.py [hunk 3]
```diff
@@ -562,11 +570,50 @@ def calc_fair_value(self) -> float:
...
+            path_payoff = np.empty(num_of_paths)
+            for path in range(num_of_paths):
+                fixing_schedule = dict(zip([0] + contractual_timeline,
+                                           np.concatenate((np.array([self._model.get_spot()]), spot_paths[path, :])) ))
+                path_payoff[path] = contract.payoff(fixing_schedule)
+            maturity = contract.get_expiry()
+            fv = mean(path_payoff) * self._model.get_df(maturity)
+            fv_contint = [(mean(path_payoff) + 1.96 * mult * np.std(path_payoff, ddof=1) / np.sqrt(self.params.num_of_paths))
+                          * self._model.get_df(maturity) for mult in [-1, 1]]
+            return fv, fv_contint
```
Comment (friedgab) [inline src/pricer.py:616]: calc_fair_value should return a single float. Greeks will not work with this implementation.

File: src/contract.py (hunk 1 above)
Comment (vargala4) [reply src/contract.py:189]: They were not introduced, I have not yet deleted them. Will do it this week.

File: src/pricer.py (hunk 2 above)
Comment (vargala4) [reply src/pricer.py:12]: Nothing, will delete it.

File: src/pricer.py (hunk 3 above)
Comment (vargala4) [reply src/pricer.py:616]: This is also the old version, will do it similar to the GenericMCPricer by defining the method calc_fair_value_with_ci, while calc_fair_value will use the one with confidence interval, but only producing the PV.

**Pull request merged**